cli/command/stack/swarm/deploy_composefile.go

```diff
@@ -249,6 +249,12 @@ func deployServices(
 				// service update.
 				serviceSpec.TaskTemplate.ContainerSpec.Image = service.Spec.TaskTemplate.ContainerSpec.Image
 			}
+
+			// Stack deploy does not have a `--force` option. Preserve existing ForceUpdate
+			// value so that tasks are not re-deployed if not updated.
+			// TODO move this to API client?
+			serviceSpec.TaskTemplate.ForceUpdate = service.Spec.TaskTemplate.ForceUpdate
```

Comment:
I was also comparing `serviceSpec` with `service.Spec` here, and noticed there's quite some other properties that differ; not sure if those are expected;

```
{"Name":"test_redis","Labels":{"com.docker.stack.image":"redis:3.2.8","com.docker.stack.namespace":"test"},"TaskTemplate":{"ContainerSpec":{"Image":"redis:3.2.8@sha256:8eb0cd22644dc1218680db876411f45baac493ac9a0bfd391d6efdef0767ea6b","Labels":{"com.docker.stack.namespace":"test"},"Privileges":{"CredentialSpec":null,"SELinuxContext":null},"Isolation":"default"},"Resources":{},"Placement":{"Platforms":[{"Architecture":"amd64","OS":"linux"}]},"Networks":[{"Target":"x5ce5v6s4fzjdd6eavt9njahb","Aliases":["redis"]}],"ForceUpdate":1,"Runtime":"container"},"Mode":{"Replicated":{"Replicas":1}},"EndpointSpec":{"Mode":"vip"}}
{"Name":"test_redis","Labels":{"com.docker.stack.image":"redis:3.2.8","com.docker.stack.namespace":"test"},"TaskTemplate":{"ContainerSpec":{"Image":"redis:3.2.8                                                                        ","Labels":{"com.docker.stack.namespace":"test"},"Privileges":{"CredentialSpec":null,"SELinuxContext":null}                      },"Resources":{},"Placement":{                                                   },"Networks":[{"Target":"test_default"             ,"Aliases":["redis"]}],"ForceUpdate":1                      },"Mode":{"Replicated":{            }},"EndpointSpec":{            }}
```